submissions/antonkottans/tjsw-post-oop/index.js

```diff
@@ -0,0 +1,140 @@
+class Creature {
+    constructor({ species, name, gender, saying, friends }) {
+        this.name = name;
+        this.gender = gender;
+        this.friends = friends;
+    }
+    returnFullInfo() {
+        const keys = [
+            "species",
+            "name",
+            "gender",
+            "legs",
+            "hands",
+            "paws",
+            "tail",
+            "saying",
+            "friends",
+        ];
+        return keys
+            .reduce((output, key) => {
+                if (this[key]) {
+                    if (typeof this[key] === "function")
+                        return output.concat(this[key]());
+                    else return output.concat(this[key]);
+                } else return output;
+            }, [])
```

Comment:
This can be optimized significantly. Too verbose, to much ifs. Swissknife `.reduce` doesn't add to readability when the solution can be implemented with `.map`, `.join` and other Array methods wise people in ECMAscript committee provided for us.

Reply:
Its a bit more verbose, but now it returns friends' names, if someone has some